### PR TITLE
Add endpoints for get keys and eds

### PR DESCRIPTION
**Pull request by jyotimahapatra**
We are creating a inhouse monitor that will leverage these endpoints and compare xdsrelay and upstream cache.

### CODE REVIEW

File: internal/app/admin/http/key.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package handler
```
Comment (jessicayuen) [inline internal/app/admin/http/key.go:1]: nit: `internal/app/admin/http/cache.go`?

File: internal/app/admin/http/key.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+func keyDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			errMessage, _ := stringify.InterfaceToString(&marshallable.Error{
+				Message: fmt.Sprintf("error in getting cache keys: %s", err.Error()),
+			})
+			_, _ = w.Write([]byte(errMessage))
```
Comment (jessicayuen) [inline internal/app/admin/http/key.go:20]: Similar comment with err handling as the EDS one.

File: pkg/marshallable/keys.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package marshallable
```
Comment (jessicayuen) [inline pkg/marshallable/keys.go:1]: `cache.go`?

File: internal/app/admin/http/eds.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+func edsDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		edsKey := getParam(req.URL.Path)
+		if edsKey == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			s, _ := stringify.InterfaceToString(&marshallable.Error{
+				Message: "Empty key",
+			})
+			_, _ = w.Write([]byte(s))
```
[the remark this answers is not in the document]
Comment (jyotimahapatra) [reply internal/app/admin/http/eds.go:21]: the linter complains about unhandled errors and the header already says bad request.

File: internal/app/admin/http/handler.go [hunk 5]
```diff
@@ -58,6 +58,18 @@ func getHandlers(bootstrap *bootstrapv1.Bootstrap,
...
+			"print the eds payload for a particular key. usage: `/eds/<key>`",
+			edsDumpHandler(orchestrator),
+			true,
+		},
+		{
+			"/keys",
+			"print all keys",
+			keyDumpHandler(orchestrator),
+			true,
+		},
```
[the remark this answers is not in the document]
Comment (jyotimahapatra) [reply internal/app/admin/http/handler.go:72]: makes sense

File: internal/app/admin/http/key.go (hunk 2 above)
Comment (jyotimahapatra) [reply internal/app/admin/http/key.go:20]: here the header is already 500, there would be no difference in making the check

**Review by jessicayuen — APPROVED** [review]

**Pull request merged**